pvsystemprofiler/tilt_azimuth_study.py

```diff
@@ -0,0 +1,281 @@
+        self.results = None
+        self.results_uncoupled = None
+        self.tilt_estimate_uncoupled = None
+        self.azimuth_estimate_uncoupled = None
+
+    def run(self):
+        self.find_boolean_daytime()
+        self.make_delta()
+        self.make_omega()
+        self.find_fit_costheta()
+        self.select_days()
+        if ~np.any(self.boolean_daytime_range):
+            print('Data in selected day_range does not meet requirements for find tilt and azimuth estimation.\n'
+                  'Please increase or shift the day range')
+            return
+        self.run_curve_fit()
+        try:
+            self.run_curve_fit_tilt_only()
+        except RuntimeError:
+            self.tilt_estimate_uncoupled = np.nan
+        try:
+            self.run_curve_fit_azimuth_only()
+        except RuntimeError:
+           self.azimuth_estimate_uncoupled = np.nan
+
+        self.estimate_costheta()
+        if self.phi_true_value is not None:
+            if self.beta_true_value is not None:
+                if self.gamma_true_value is not None:
+                    self.ground_truth_costheta()
+                    self.results = pd.DataFrame(columns=['Latitude Residual', 'Tilt Residual',
+                                                 'Azimuth Residual'])
+                    r1 = self.phi_true_value - self.latitude_estimate
+                    r2 = self.beta_true_value - self.tilt_estimate
+                    r3 = self.gamma_true_value - self.azimuth_estimate
+                    self.results.loc[0] = [r1, r2, r3]
+
+                    # uncoupled tilt and azimuth results
+                    self.results_uncoupled = pd.DataFrame(columns=['Latitude Residual', 'Tilt Residual',
+                                                 'Azimuth Residual'])
+                    r1 = self.phi_true_value - self.latitude_estimate
+                    r2 = self.beta_true_value - self.tilt_estimate_uncoupled
+                    r3 = self.gamma_true_value - self.azimuth_estimate_uncoupled
+                    self.results_uncoupled.loc[0] = [r1, r2, r3]
+
+        return
+
+    def find_boolean_daytime(self):
+        if self.daytime_threshold is None:
+            self.boolean_daytime = np.zeros([self.daily_meas, self.num_days], dtype=bool)
+            self.daytime_threshold_fit = self.find_daytime_threshold_quantile_seasonality()
+            self.boolean_daytime = self.data_matrix > self.daytime_threshold_fit
+        else:
+            self.boolean_daytime = find_daytime(self.data_matrix, self.daytime_threshold)
+
+    def make_delta(self):
+        """Delta is estimated  using equation (1.6.1a) in:
+        Duffie, John A., and William A. Beckman. Solar engineering of thermal
+        processes. New York: Wiley, 1991."""
+        delta_1 = np.deg2rad(23.45 * np.sin(np.deg2rad(360 * (284 + self.day_of_year) / 365)))
+        self.delta = np.tile(delta_1, (self.daily_meas, 1))
+        return
+
+    def make_omega(self):
```

Comment:
_Same as delta (declination), we should make a separate utility module for this equation: `utilites/hour_angle_equation.py`.

Also, I don't believe this citation or equation is correct. In my version of Duffie, there is no equation (1.6.1), only the pair of equations (1.6.1a) and (1.6.1b) that govern the declination, delta. Omega is defined on p. 13 as, "Hour angle, the angular displacement of the sun east or west of the local meridian due to rotation of the earth on its axis at 15° per hour; morning negative, afternoon positive." Solar time is related to clock time by equation (1.5.2). So, the calculation for omega must depend on the equation of time. Currently, this is being calculated on _clock time_ not _solar time_.

Reply:
So, when this is moved into it's own file, it should import from `utilities.equation_of_time` for its calculation.